ring-jetty-adapter/src/ring/adapter/jetty.clj

```diff
@@ -28,11 +28,37 @@
           (servlet/update-servlet-response response response-map)
           (.setHandled base-request true))))))
 
-(defn- ^AbstractHandler async-proxy-handler [handler timeout]
+(defn- attach-async-timeout-listener
+  [^HttpServletRequest request
+   ^AsyncContext context
+   ^HttpServletResponse response
+   timeout-handler]
+  (.addListener
+   context
+   (proxy [AsyncListener] []
+     (onTimeout [^AsyncEvent event]
+       (timeout-handler
+         (servlet/build-request-map request)
+         (fn [response-map]
+           (servlet/update-servlet-response response context response-map))
+         (fn [^Throwable exception]
```

Comment:
This could be factored out into its own function, since we're using it twice.

```clojure
(defn- jetty-raise [^AsyncContext context ^HttpServletResponse response]
  (fn [^Throwable ex]
    ...))
```